test/types/models.test.ts

```diff
@@ -216,6 +216,7 @@ function find() {
   Project.find({ name: 'Hello' });
 
   // just callback
+  // @ts-expect-error: Callback to find is longer supported
   Project.find((error: CallbackError, result: IProject[]) => console.log(error, result));
```

Comment:
i dont know why no `expectError` was here previously (maybe because `any` somehow allowing it?), but maybe remove the comment again

Reply:
Ah sorry, forgot to update this one, thanks

Yeah the type `{ [key: string]: any }` (which is what `FilterQuery` basically reduced to before) is super permissive, so a function actually passes it (as well as `ObjectId` and `Query` which were supposed to be valid, but weren't very explicitly typed to be valid)